flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/asyncprocessing/AbstractAsyncStateStreamOperator.java

```diff
@@ -179,9 +181,12 @@ public final <T> ThrowingConsumer<StreamRecord<T>, Exception> getRecordProcessor
 
     /** Create new state (v2) based on new state descriptor. */
     protected <N, S extends State, T> S getOrCreateKeyedState(
-            TypeSerializer<N> namespaceSerializer, StateDescriptor<T> stateDescriptor)
```

Comment:
Should this function override the `AbstractStreamOperator#getOrCreateKeyedState`? I am worried that developers may misuse it. When writing asynchronous operators, calling `getOrCreateKeyedState` without `defaultNamespace` ​​parameter would create a synchronous state.

Reply:
Well, they have different signatures, as the `StateDescriptor` of state v2 and v1 are different. But in future, we could override the original one and throw `UnsupportedException`, after the state v2 is fully functional.